Wait Step

umerkhan-apple opened the pull request:
### Wait Step
- Make ORKWaitStep - public, make ORKWaitStepViewController - private, make ORKWaitStepView - project.
- __ORKDefines.h__
    - Refactor ORKProgressIndicatorMask into ORKProgressIndicatorType and change from NS_OPTIONS to NS_ENUM.
        - Main reason for this is because we don’t want to allow more than one option at a time.
- __Localizable.strings__
    - Remove the localizable string because we don’t want a fallback text. If the user did not specify a title and text on the step, then it should just show the indicator type.
        - A use case could be to just show the activity indicator view without a message.
- __ORKWaitStep.m__
    - Override `allowsBackNavigation` to disable it because we don’t want the user to be able to go back while the background activity is begin run.
    - Add `copyWithZone` and `isEqual`
- __ORKWaitStepView.h__
    - Remove all properties from .h file except progressView, since that is the only one we need to be exposed.
    - Make progressView - readonly in the .h file.
    - Remove the extra imports from .h file.
    - Update legal header
    - Add extra lines around imports.
    - Change init method because we don’t want a heading anymore.
        - The reason we don’t want a heading is because ORKVerticalContainerView already has a headerView which contains a label. That is where the title and text provided from the STEP should be assigned.
- __ORKWaitStepView.m__
    - Add extra lines around imports
    - Create local variables for the ones we removed from .h file.
    - Instead of adding the progress view to self, add it to self.stepView.
        - StepView is a place where additional UI components should go for an ORKVerticalContainerView class.
    - Also set verticalCenteringEnabled = YES.
    - We don’t need to check for progressView and activituiyindicatorview and remove them from superview because we shouldn’t allow the option to change the indicator type once you have specified it.
        - So consequently, we can just add them to the stepView using a switch statement, and not have to worry about removing them.
    - Since we no longer need the textLabel, we can remove it completely from this file.
        - This includes the constraints and the accessibility.
        - Those will be taken care of by ORKVerticalContainerView which includes the headerView for a title and a text.
- __ORKWaitStepViewController.h__
    - We want to remove setCurrentProgressIndicatorMask, since we don’t want to allow that use case.
    - Refactor setProgressDescription to updateText.
        - This is because we are going to be updating the text property on the headerView of ORKVerticalContainerView.
- __ORKWaitStepViewController.m__
    - Create a method to return waitStep (already casted in the method, so you don’t have to do it again and again).
    - Then just create a waitStepView, assign the title and text from the step to its headerView labels.
    - In constraints, just allow waitStepView to cover horizontally and vertically.
- __TaskListRow.swift__
    - Just assign the global `exampleQuestionText` to step’s title property and `exampleDescription` to step’s text property.
- __TaskViewController.swift__
    - Update the timer interval to be 0.1, and increase progress by 0.01 to give it a smoother progress bar look.
        - Currently, it looks a little fidgety.
- __MainViewController.m__
    - Change the selector for the indeterminateStep to be goForward instead of finish.
        - Finish is only used for ORKActiveStep. Since you refactored to ORKStep, the finish is no longer there, which is causing a crash.
    - Assign text and title to each wait step.
    - Update the progress every 0.01 and change the text when progress is half way done.
        - When the progress is done, call goForward instead of finish.

codecadwallader commented:
@umerkhan-apple Thanks Umer for these improvements.  I did notice one minor issue where the caching of the percent formatter was lost (requested by Yuan).  I'll re-apply it to the branch.

umerkhan-apple commented:
@codecadwallader Yes, I'm glad you caught that!

I just updated that branch also and created a new PR. It includes apple's attribution for copyright. Can you please merge that as well? 

Also note that ORKWaitStepView.h and ORKWaitStepView.m did not have the correct legal header. I have added the RK default header. Please modify as necessary for you.

Once, you do the above listed two things. I'll just make sure everything went smoothly in the merges and that nothing is failing or faltering, and we should be good to go with this PR.
